watcher/watcher.go

```diff
@@ -1068,30 +1068,73 @@ func (w *Watcher) EnsureCorrectRooms() error {
 		logger.Info("no invalid pods to delete")
 	}
 
-	for _, podName := range podsToDelete {
-		err = controller.DeletePodAndRoom(logger, w.RoomManager, w.MetricsReporter,
-			k, w.RedisClient.Client, configYaml, podName, reportersConstants.ReasonInvalidPod)
-		if err != nil {
-			logger.WithError(err).WithField("podName", podName).Error("failed to delete pod")
-		}
+	timeoutSec := w.Config.GetInt("updateTimeoutSeconds")
+	timeoutDur := time.Duration(timeoutSec) * time.Second
+	willTimeoutAt := time.Now().Add(timeoutDur)
+
+	// Lock down scaling so it doesn't interfer with rolling update surges
+	downScalingLockKey := models.GetSchedulerDownScalingLockKey(w.Config.GetString("watcher.lockKey"), configYAML.Name)
+	downScalingLock, _, err := controller.AcquireLock(
+		ctx,
+		logger,
+		w.RedisClient,
+		w.Config,
+		nil,
+		downScalingLockKey,
+		w.SchedulerName,
+	)
+	defer controller.ReleaseLock(
+		logger,
+		w.RedisClient,
+		downScalingLock,
+		w.SchedulerName,
+	)
+
+	if err != nil {
```

Comment:
doesn't it make more sense to check err before deferring ReleaseLock?

Reply:
I think ReleaseLock is a noop if lock is nil, so it doesn't matter. But correct me if I am wrong @lftakakura

Reply:
@cscatolini it's right. It makes no difference as ReleaseLock can be called when a lock is not acquired